src/options.ts

```diff
@@ -66,7 +94,48 @@ function addCountdown(builder: PanelOptionsEditorBuilder<ClockOptions>) {
         placeholder: 'ISO 8601 or RFC 2822 Date time',
       },
       defaultValue: dateTime(Date.now()).add(6, 'h').format(),
-      showIf: (o) => o.mode === ClockMode.countdown,
+      showIf: (o) => o.mode === ClockMode.countdown && o.countdownSettings.source === ClockSource.input,
+    })
+    .addSelect({
+      category,
+      path: 'countdownSettings.queryCalculation',
+      name: 'Calculation',
+      description: 'How to calculate the countdown time',
+      settings: {
+        options: [
+          {
+            value: CountdownQueryCalculation.lastNotNull,
+            label: 'Last *',
+            description: 'Last non-null value (also excludes NaNs)',
+          },
+          { value: CountdownQueryCalculation.last, label: 'Last', description: 'Last value' },
+          {
+            value: CountdownQueryCalculation.firstNotNull,
+            label: 'First *',
+            description: 'First non-null value (also excludes NaNs)',
+          },
+          { value: CountdownQueryCalculation.first, label: 'First', description: 'First value' },
+          { value: CountdownQueryCalculation.min, label: 'Min', description: 'Minimum value' },
+          {
+            value: CountdownQueryCalculation.minFuture,
+            label: 'Min Future',
+            description: 'Minimum value that is in the future',
+          },
+          { value: CountdownQueryCalculation.max, label: 'Max', description: 'Maximum value' },
+        ],
+      },
+      defaultValue: CountdownQueryCalculation.last,
+      showIf: (o) => o.mode === ClockMode.countdown && o.countdownSettings.source === ClockSource.query,
+    })
+    .addSelect({
+      category,
+      path: 'countdownSettings.queryField',
+      name: 'Field',
+      settings: {
+        options: data?.[0]?.fields?.map((f: any) => ({ label: f.name, value: f.name })) ?? [],
```

Comment:
You can use instead `addFieldNamePicker` from `grafana-data` (and in the count up)